src/Nethermind/Ethereum.Test.Base/GeneralTestBase.cs

```diff
@@ -103,7 +104,13 @@ protected EthereumTestResult RunTest(GeneralStateTest test, ITxTracer txTracer)
             header.Hash = header.CalculateHash();
 
             Stopwatch stopwatch = Stopwatch.StartNew();
-            transactionProcessor.Execute(test.Transaction, header, txTracer);
+            var txValidator = new TxValidator((MainnetSpecProvider.Instance.ChainId));
+            var spec = specProvider.GetSpec(test.CurrentNumber);
+            if (test.Transaction.ChainId == null)
+                test.Transaction.ChainId = MainnetSpecProvider.Instance.ChainId;
+            bool isValid = txValidator.IsWellFormed(test.Transaction, spec);
+            if (isValid)
```

Comment:
I think you need to throw on invalid? or not?

Reply:
I think not, Ethereum tests are passing now, this was actually added by @MarekM25

Reply:
Another solution is adding TxValidator logic in TransactionProcessor. What do you think @tkstanczak @LukaszRozmej ?

Reply:
ignore